robot/impl/local_robot.go

```diff
@@ -1204,6 +1204,30 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 		return
 	}
 
+	r.logger.CInfo(ctx, "running module setup phase")
+	// Diff configs before resolving dependencies to find what modules have 
+	// been added or updated. We run the setup phase for these modules before 
+	// proceeding with reconfiguration.
+	diff, err := config.DiffConfigs(*r.Config(), *newConfig, r.revealSensitiveConfigDiffs)
+	if err != nil {
+		r.logger.CErrorw(ctx, "error diffing the configs for setup phase", "error", err)
+		return
+	}
+	// TODO: merge Added + Modified into one loop
+	for _, mod := range diff.Added.Modules {
+		if err := r.manager.moduleManager.FirstRun(ctx, mod); err != nil {
```

Comment:
first run should be run in parallel - let's make this a fast follow but we shouldn't lose track

Reply:
i had this thought also and yeah it should at least be a fast-follow if not part of this PR, since the setup phase commands we are expecting will be slow (`docker compose`, etc). the setup phase logic owned by RDK should not be mutating any shared fields so operation should be thread-safe.

i suppose we need to communicate to the user that setup phases will be run in parallel, so scripts should be thread safe. alternatively we can expose an option that dictates whether to setup concurrently or serially

Reply:
discussed IRL - it's okay to have setup scripts run concurrently - we just need to document this behavior clearly. it's okay since we already do module/resource processing concurrently. the main difference, at least w.r.t resources, is that users cannot specific dependency chains with setup scripts, but they can get around this by bundling all setup logic into a single script.